compiler/src/dotty/tools/dotc/printing/Formatting.scala

```diff
@@ -71,6 +71,16 @@ object Formatting {
       given Show[TypeComparer.ApproxState] with
         def show(x: TypeComparer.ApproxState) = TypeComparer.ApproxState.Repr.show(x)
 
+      given Show[ast.TreeInfo.PurityLevel] with
+        def show(x: ast.TreeInfo.PurityLevel) = x match
+          case ast.TreeInfo.Path           => "PurityLevel.Path"
+          case ast.TreeInfo.Pure           => "PurityLevel.Pure"
+          case ast.TreeInfo.Idempotent     => "PurityLevel.Idempotent"
+          case ast.TreeInfo.Impure         => "PurityLevel.Impure"
+          case ast.TreeInfo.PurePath       => "PurityLevel.PurePath"
+          case ast.TreeInfo.IdempotentPath => "PurityLevel.IdempotentPath"
+          case _                           => s"PurityLevel(${x.x})"
```

Comment:
I was looking at 

```scala
        val countsAsPure =
          if dropOp(tree1).symbol.isInlineVal
          then isIdempotentExpr(tree1)
          else isPureExpr(tree1)
```

in `constToLiteral` and needed to print purity levels.

Reply:
Could this be achieved by making PurityLevel an enum instead to generate a sensible toString?

Reply:
Perhaps.  But what's a nice way to reimplement `min`?

Reply:
Ah I hadn't considered that, nevermind then.

Reply:
This could still be implemented as the toString of PurityLevel since it doesn't require a Context.

Reply:
Sure, it could.